### ISSUE TITLE
[chromecast] Installing Chromecast Binding in 2.5.0.M2 breaks REST UI

### ISSUE THREAD

**Issue by Borcon** [openhab/openhab2-addons#5986]
If you install the chromecast binding in m2 then the rest ui api website stops working.
Error 500
![image](https://user-images.githubusercontent.com/39000945/63746004-ef358f80-c8a3-11e9-9f82-1fee7ccaddf1.png)

## Your Environment
Raspberry Pi3,
openhab m2

**Comment by wborn** [comment]
Any binding that installs Jackson 2.9.x as dependency breaks the REST Docs UI (Swagger), see also:
* https://github.com/openhab/openhab2-addons/issues/5742
* https://github.com/openhab/openhab-webui/issues/96
* https://github.com/openhab/openhab-webui/issues/67

**Comment by meingraham** [comment]
@wborn Is it possible to install an older Chromecast binding jar? Would I be missing any dependencies that I would also have to install? If so, which repository? Where in the Jenkins repository (https://ci.openhab.org/) would I find it?

I'm am beginning the migration of my openHABian deployment to mqtt2. It required an upgrade from 2.4.0-1. However, it doesn't work with the latest Snapshot and I had to downgrade to M2. I need the REST Docs to ease the migration as I can use the UI to clone my mqtt2 Things easily. Even though the REST API works, that would just complicate things and I ,might as well use PaperUI at that point. So, I'd really, really like to get the REST Docs working along with mqtt2. :wink:

**Comment by wborn** [comment]
I haven't tried that myself but sometimes it works. You could give it a try using these:

* [org.openhab.binding.chromecast-2.4.0.jar](https://bintray.com/openhab/mvn/download_file?file_path=online-repo%2F2.4%2Forg%2Fopenhab%2Fbinding%2Forg.openhab.binding.chromecast%2F2.4.0%2Forg.openhab.binding.chromecast-2.4.0.jar)
* [org.openhab.binding.chromecast-2.5.0.M1.jar](https://openhab.jfrog.io/openhab/online-repo-milestone/2.5/org/openhab/binding/org.openhab.binding.chromecast/2.5.0.M1/org.openhab.binding.chromecast-2.5.0.M1.jar)

**Comment by meingraham** [comment]
Yup, that didn't resolve it.

Perhaps it's the litvak library that's using the incompatible Jackson library version.

**Comment by meingraham** [comment]
@wborn It's the litvak library that has the Jackson dependency:
```
openhab> bundle:list -s | grep chromecast
240 │ Active │  80 │ 2.5.0.M1              │ org.openhab.binding.chromecast
277 │ Active │  80 │ 0.11.0                │ su.litvak.chromecast.api-v2
openhab> bundle:requirements 240 | grep jackson
openhab> bundle:requirements 277 | grep jackson
osgi.wiring.package; (&(osgi.wiring.package=com.fasterxml.jackson.annotation)(&(version>=2.9.0)(!(version>=3.0.0)))) resolved by:
   osgi.wiring.package; com.fasterxml.jackson.annotation 2.9.8 from com.fasterxml.jackson.core.jackson-annotations_2.9.8 [203]
osgi.wiring.package; (&(osgi.wiring.package=com.fasterxml.jackson.core)(&(version>=2.9.0)(!(version>=3.0.0)))) resolved by:
   osgi.wiring.package; com.fasterxml.jackson.core 2.9.8 from com.fasterxml.jackson.core.jackson-core_2.9.8 [205]
osgi.wiring.package; (&(osgi.wiring.package=com.fasterxml.jackson.databind)(&(version>=2.9.0)(!(version>=3.0.0)))) resolved by:
   osgi.wiring.package; com.fasterxml.jackson.databind 2.9.8 from com.fasterxml.jackson.core.jackson-databind_2.9.8 [206]
osgi.wiring.package; (&(osgi.wiring.package=com.fasterxml.jackson.databind.annotation)(&(version>=2.9.0)(!(version>=3.0.0)))) resolved by:
   osgi.wiring.package; com.fasterxml.jackson.databind.annotation 2.9.8 from com.fasterxml.jackson.core.jackson-databind_2.9.8 [206]
```

I couldn't find the M1 litvak bundle. Could I ask a favor for the URL to that bundle?

Much appreciated!

Mike

**Comment by wborn** [comment]
You can download the JARs from Maven Central:

https://search.maven.org/search?q=g:su.litvak.chromecast%20AND%20a:api-v2&core=gav

---

It's also possible to install such bundles from Maven Central on the Karaf Console using GAV (group, artifact, version) parameters as follows:

`bundle:install mvn:su.litvak.chromecast/api-v2/0.11.1`

After that you'll probably also need to start it using `bundle:start <bundle id>`

**Comment by meingraham** [comment]
Thanks Wouter.

I've decided to put my mqtt2 migration on hold until the next OH milestone where I hope mqtt2, chromecast, REST Docs, Jackson, etc. are all playing together nicely. With all my testing and reverting of bundles trying to solve the REST Docs issue, I obviously screwed something up because over the holiday weekend my openHAB server rebooted due to a power outage and openHAB did not restart successfully. I had to go through an M2-1 re-install to get it to start up. So, motto - if it ain't broke, don't fix it. And right now mqtt1 ain't broke. So, wait, I will :wink:

Cheers!

Mike
